Dockerfile

```diff
@@ -0,0 +1,68 @@
+# #####################################################################
+# # Stage 1: Install gems and precompile assets.
+# #####################################################################
+FROM ruby:3.3-slim AS build
+WORKDIR /app
+
+# Set a random secret key base so we can precompile assets.
+ENV SECRET_KEY_BASE airport_gap_secret_key_base
+
+# Install necessary dependencies to set up Node.js and Yarn repos.
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+  curl \
+  gnupg \
+  build-essential \
+  libpq-dev
+
+# Set up Node.js and Yarn package repositories.
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash -
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+
+# Install necessary packages to build gems and assets.
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+  nodejs \
+  yarn
+
+# Install gems into the vendor/bundle directory in the workspace.
+COPY Gemfile Gemfile.lock /app/
+RUN bundle config set --local path "vendor/bundle" && \
+  bundle install --jobs 4 --retry 3
+
+# Install JavaScript dependencies to precompile assets.
+# COPY package.json yarn.lock /app/
```

Comment:
do we need to uncomment this or is it not needed?

Reply:
package.json and yarn.lock don't exist, so copying them in won't have any effect, and yarn install will fail looking for them.

If there are any javascript dependencies to be loaded, we'd want to make the package.json and use yarn to generate the lock. I didn't see any place where dependencies were loaded using the old javascript libraries.

Reply:
I think all the js is local / custom, so I think this is ok